### PR TITLE
fetch-release.sh: fix BUILD_IT_DO_NOT_DOWNLOAD_IT mode

### PR DESCRIPTION
**Pull request by t-nelson**
`BUILD_IT_DO_NOT_DOWNLOAD_IT` clearly hasn't been used in awhile...

### CODE REVIEW

**Comment by tigarcia** [comment]
What was the issue with this mode? We're using it now in buildkite and it seems to be working well.

**Comment by t-nelson** [comment]
> What was the issue with this mode? We're using it now in buildkite and it seems to be working well.

can you link me (offline) to a build where it was used?  if anything fails it just blow through

File: fetch-release.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e
```
Comment (t-nelson) [inline fetch-release.sh:3]: or else we just fail silently

File: fetch-release.sh [hunk 2]
```diff
@@ -8,10 +10,14 @@ REPO=https://github.com/solana-labs/stake-o-matic
 #DEFAULT_TO_MASTER=1
 
 if [[ -n $BUILD_IT_DO_NOT_DOWNLOAD_IT ]]; then
+  if ! which cargo &>/dev/null; then
```
Comment (t-nelson) [inline fetch-release.sh:13]: buildkite user session doesn't load ~/.profile by default and we need cargo env

File: fetch-release.sh [hunk 3]
```diff
@@ -8,10 +10,14 @@ REPO=https://github.com/solana-labs/stake-o-matic
...
+    # shellcheck source=/dev/null
+    source ~/.profile
+  fi
   echo "Building locally"
   cargo build --release
   cp ./target/release/solana-stake-o-matic .
-  solana-stake-o-matic --version
+  ./solana-stake-o-matic --version
```
Comment (t-nelson) [inline fetch-release.sh:20]: the rest of the scripts expect the `solana-stake-o-matic` bin to be in `PWD`

**Review by alex-at-planetariummusic — APPROVED** [review]

**Review by tigarcia — APPROVED** [review]

**Pull request merged**